app/main.py

```diff
@@ -1,3 +1,45 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: (int, float)) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: (int, float, Distance)) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: (int, float)) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: (int, float)) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: (int, float)) -> bool:
+        return self.km < other
+
+    def __gt__(self, other: (int, float)) -> bool:
+        return self.km > other
+
+    def __eq__(self, other: (int, float)) -> bool:
+        return self.km == other
+
+    def __le__(self, other: (int, float)) -> bool:
+        return self.km <= other
+
+    def __ge__(self, other: (int, float)) -> bool:
+        return self.km >= other
```

Comment:
same. you didn't type checked.
you can write this methods in one line even with type checks. Just use previously created magic methods

Reply:
Yes, I know. For example 
return not self.__lt__(other.km) if isinstance(other, Distance) else not self.__lt__(other)
I fixet it.
Please check

Reply:
> same. you didn't type checked. you can write this methods in one line even with type checks. Just use previously created magic methods

I'm not that sure )
![image](https://user-images.githubusercontent.com/107141441/217650076-ab80a266-676a-4dc1-8398-1a1cac61df20.png)